infra/modules/providers/azure/api-mgmt/main.tf

```diff
@@ -0,0 +1,23 @@
+module "azure-provider" {
+    source = "../provider"
+}
+
+resource "azurerm_resource_group" "apimgmt" {
+  name     = "${var.resource_group_name}"
+  location = "${var.resource_group_location}"
+  tags     = "${var.resource_tags}"
+}
+
+resource "azurerm_api_management" "apimgmt" {
+  name                = "${var.apimgmt_name}"
```

Comment:
so how does a developer configure backend api's using this module?

Reply:
This is an [example](https://github.com/Microsoft/bedrock/blob/530830d09ccb43e3756dea839981530050ed287b/cluster/environments/azure-api-mgmt/push_api_mgmt.sh) of how bedrock does it using null_resource and running the script. This is a limitation in terraform

Reply:
It would be nice if we can deploy the endpoints using the Terraform way, instead of using a shell script.

@erikschlegel If we can update the Azure rm provider that comes with Terraform (currently .1.23) to 1.24 or later we could support api creations and just pass the configuration as variables.  https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/CHANGELOG.md